scripts/bootstrap-py2-venv

```diff
@@ -0,0 +1,53 @@
+# This script is meant to be sourced by bash, not executed.
+# As such, don't use exit, use return.
+
+# This script is purely a python virtualenv bootstrap, not the entire development environment.
+# For the entire thing, you should still use make bootstrap.
+# Eventually, the goal would be to have an idempotent single script that does
+# everything make bootstrap does and more (system dependencies and so on).
+
+# This... doesn't quite work.
+# set -e
+# trap "echo 'bootstrap FAILED.' && trap - ERR && return" ERR
+
+# So just going to sprinkle returns everywhere.
+
```

Comment:
Let's do something like this around here https://superuser.com/a/1288646

```suggestion
if [ "$0" = "$BASH_SOURCE" ]; then
    echo "Error: Please run `source $0` instead of `./$0`.
    exit 1
fi
```